mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/SmartTestingMavenConfigurer.java

```diff
@@ -133,16 +139,56 @@ private void calculateChanges(File projectDirectory, Configuration configuration
-            if (wasConfigured) {
-                configuration.dump(mavenProject.getBasedir());
-                if (isFailedStrategyUsed()) {
-                    SurefireReportStorage.copySurefireReports(mavenProject.getModel());
-                }
+        final Consumer<MavenProject> configureSmartTestingExtensionAction;
+        final File executionRootDirectory = new File(session.getExecutionRootDirectory());
+        if (hasModuleSpecificConfigurations(executionRootDirectory, this::isProjectRootDirectory)) {
+            configureSmartTestingExtensionAction = applyModuleSpecificConfiguration(configuration);
+        } else {
+            final MavenProjectConfigurator mavenProjectConfigurator = new MavenProjectConfigurator(configuration);
+            configureSmartTestingExtensionAction = mavenProject -> configureMavenProject(mavenProjectConfigurator, mavenProject, configuration);
+        }
+        session.getAllProjects().forEach(configureSmartTestingExtensionAction);
+    }
+
+    private boolean hasModuleSpecificConfigurations(File projectDir, Function<File, Boolean> stopCondition) {
+        final ConfigLookup lookUp = new ConfigLookup(projectDir, stopCondition);
+        final boolean moreThanOneConfigFile = lookUp.hasMoreThanOneConfigFile(SMART_TESTING_YML, SMART_TESTING_YAML);
+        return moreThanOneConfigFile && System.getProperty(SMART_TESTING_CONFIG) == null;
+    }
+
+    private Consumer<MavenProject> applyModuleSpecificConfiguration(Configuration configuration) {
+        return mavenProject -> {
+            final ConfigLookup configLookup = new ConfigLookup(mavenProject.getBasedir(), this::isProjectRootDirectory);
+            final File dirWithConfig = configLookup.getFirstDirWithConfigOrWithStopCondition();
+            final Configuration mavenProjectConfiguration = configLookup.isConfigFromProjectRootDir() ? configuration :
+                ConfigurationLoader.load(dirWithConfig);
+            if (mavenProjectConfiguration.isDisable()) {
+                logExtensionDisableReason(logger,
+                    SMART_TESTING_DISABLE + " is set for module " + mavenProject.getArtifactId());
+                return;
             }
-        });
+            if (mavenProjectConfiguration.areStrategiesDefined()) {
+                final MavenProjectConfigurator mavenProjectConfigurator =
+                    new MavenProjectConfigurator(mavenProjectConfiguration);
+                configureMavenProject(mavenProjectConfigurator, mavenProject, mavenProjectConfiguration);
+            } else {
+                logger.warn(
+                    "Smart Testing Extension is installed but no strategies are provided for %s module. It won't influence the way how your tests are executed. "
+                        + "For details on how to configure it head over to http://bit.ly/st-config",
+                    mavenProject.getArtifactId());
+            }
```

Comment:
maybe, it would be good to have the checks (if the ST should be installed for the module) in separated class. 
As it is done for [SkipInstallationChecker](https://github.com/arquillian/smart-testing/blob/master/mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/SkipInstallationChecker.java) as well as for [SkipModuleChecker](https://github.com/arquillian/smart-testing/blob/master/mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/SkipModuleChecker.java)

Reply:
I think we are just checking `configuration.isDisable()` to check to ST should be installed for module. So I don't think we should move it to separated class.

Reply:
You check two things here: 
`mavenProjectConfiguration.isDisable()`
and
`mavenProjectConfiguration.areStrategiesDefined()`